### PR TITLE
24549 Added missing NR request types

### PR DESCRIPTION
**Pull request by severinbeauvais**
*Issue #:* bcgov/entity#24549

*Description of changes:*
- added missing accepted request types
- added comments
- updated an error message
- moved request types together in mapping array
- added some missing types
- fixed and expanded unit tests

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the lear license (Apache 2.0).

### CODE REVIEW

**Comment by severinbeauvais** [comment]
I'm still working through the unit tests, but please review the code changes and lmk if OK.

File: legal-api/src/legal_api/resources/v1/nr_type_map.py
Comment (vysakh-menon-aot) [inline legal-api/src/legal_api/resources/v1/nr_type_map.py:1]: Not really required. We don't use any endpoints in v1

**Review by vysakh-menon-aot — APPROVED** [review]

File: legal-api/src/legal_api/resources/v1/nr_type_map.py
Comment (severinbeauvais) [reply legal-api/src/legal_api/resources/v1/nr_type_map.py:1]: Yes, this may be an obsolete endpoint. Shall I retain my change or revert it?

**Review by ArwenQin — APPROVED** [review]
The code changes look good

File: legal-api/tests/unit/services/filings/validations/test_alteration.py [hunk 1]
```diff
@@ -154,7 +190,7 @@ def test_alteration(session, use_nr, new_name, legal_type, new_legal_type, nr_ty
 
         f['filing']['alteration']['nameRequest']['nrNumber'] = identifier
         f['filing']['alteration']['nameRequest']['legalName'] = new_name
-        f['filing']['alteration']['nameRequest']['legalType'] = legal_type
+        f['filing']['alteration']['nameRequest']['legalType'] = new_legal_type
```
Comment (severinbeauvais) [inline legal-api/tests/unit/services/filings/validations/test_alteration.py:193]: So... this and line 204, below, were the reasons my initial tests failed 🤦‍♂️

File: legal-api/tests/unit/services/filings/validations/test_alteration.py [hunk 2]
```diff
@@ -244,7 +302,7 @@ def test_validate_numbered_name(session, test_name, legal_type, new_legal_type,
         # check that validation failed
         assert err
         assert HTTPStatus.BAD_REQUEST == err.code
-        assert err.msg[0]['error'] == err_msg
+        assert err_msg in err.msg[0]['error']
```
Comment (severinbeauvais) [inline legal-api/tests/unit/services/filings/validations/test_alteration.py:305]: I changed this so we don't have to use the full, 2-line, error messages in the test parameters.

**Comment by severinbeauvais** [comment]
@vysakh-menon-aot Are you OK with my latest changes (second commit)?

**Pull request merged**